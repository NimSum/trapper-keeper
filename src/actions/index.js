export const addNotes = notes => ({
  type: 'ADD_NOTES',
  notes
});

export const addNote = note => ({
  type: 'ADD_NOTE',
  note
})

export const updateNote = note => ({
  type: 'UPDATE_NOTE',
  note
})
