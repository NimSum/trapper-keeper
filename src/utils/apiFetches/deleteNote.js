export const deleteNote = async(id)=>{
    const url = `http://localhost:3000/api/v1/notes/${id}`;
    const response = await fetch(url, {
        method: 'DELETE'
    })
    return response;
}