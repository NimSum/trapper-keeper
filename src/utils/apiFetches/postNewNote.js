export const postNewNote = async (note) => {
  const url = 'http://localhost:3000/api/v1/notes';
  const response = await fetch(url, {
    method: "POST",
    body: JSON.stringify(note),
    headers: {
      "Content-type": "application/json"
    }
  });

  if (!response.ok) {
    throw Error('Failed to post new note');
  }

  const result = await response.json();
  return result;
}