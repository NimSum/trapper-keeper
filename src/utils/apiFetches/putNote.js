export const putNote = async (note) => {
  const url = `http://localhost:3000/api/v1/notes/${note.id}`;
  const response = await fetch(url, {
    method: 'PUT',
    body: JSON.stringify(note),
    headers: {
      "Content-type": "application/json"
    }
  })

  if (!response.ok) {
    throw Error('Failed to update note');
  }

  return response;
}