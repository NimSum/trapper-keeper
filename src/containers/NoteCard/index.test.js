import React from 'react';
import { NoteCard, mapDispatchToProps } from './index';
import { shallow } from 'enzyme';
import { updateNote, deleteNote } from '../../actions/';
import { deleteNoteFetch } from '../../utils/apiFetches/deleteNote';
import { putNote } from '../../utils/apiFetches/putNote';

describe('Notecard container', () => {
  const mockNoteCard = {
    title: "Mock Note",
    id: "1",
    listItems: [
      { id: "1", body: "take out trash", completed: false },
      { id: "2", body: "wash dishes", completed: false }
    ]
  }
  let mockUpdateExistingNote = jest.fn();
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      < NoteCard 
        note={ mockNoteCard }
        updateExistingNote={ mockUpdateExistingNote }/>
    )
  })

  it('should have default state', () => {
    expect(wrapper.state()).toHaveProperty('id');
    expect(wrapper.state()).toHaveProperty('listItems');
    expect(wrapper.state()).toHaveProperty('title');
  })

  it('should match component snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe('updateListItems', () => {
    const updatedListItem = { 
      id: "1", 
      body: "take out trash in kitchen", 
      completed: true 
    }
    const expected = [
      updatedListItem,
      { id: "2", body: "wash dishes", completed: false }      
    ]
    it('should update listItem within local state', () => {
      wrapper.instance().updateListItems(updatedListItem);
      expect(wrapper.state().listItems).toEqual(expected);
    })
  })
})