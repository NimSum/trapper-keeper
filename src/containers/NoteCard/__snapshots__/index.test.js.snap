// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notecard container should match component snapshot 1`] = `
<NavLink
  activeClassName="active"
  exact={true}
  style={
    Object {
      "textDecoration": "none",
    }
  }
  to="/notes/1"
>
  <article
    className="note-card"
    style={
      Object {
        "background": "white",
      }
    }
  >
    <h3>
      Mock Note
    </h3>
    <ul
      className="note-items"
    >
      <ListItem
        item={
          Object {
            "body": "take out trash",
            "completed": false,
            "id": "1",
          }
        }
        key="1"
        updateListItems={[Function]}
      />
      <ListItem
        item={
          Object {
            "body": "wash dishes",
            "completed": false,
            "id": "2",
          }
        }
        key="2"
        updateListItems={[Function]}
      />
      <p>
        Completed
      </p>
      <hr />
    </ul>
    <div
      className="svg-container"
    >
      <div
        className="color-svgs"
      >
        <i
          className="fas fa-circle white"
          data-color="white"
          onClick={[Function]}
        />
        <i
          className="fas fa-circle blue"
          data-color="#40c3ff"
          onClick={[Function]}
        />
        <i
          className="fas fa-circle red"
          data-color="#ff6d3f"
          onClick={[Function]}
        />
        <i
          className="fas fa-circle yellow"
          data-color="#fcf296"
          onClick={[Function]}
        />
      </div>
      <div>
        <i
          className="fas fa-trash-alt"
          onClick={[Function]}
        />
      </div>
    </div>
  </article>
</NavLink>
`;
